src/Compilers/Core/Portable/Compilation/SyntaxTreeOptionsProvider.cs

```diff
@@ -14,9 +14,14 @@ public abstract class SyntaxTreeOptionsProvider
         public abstract bool? IsGenerated(SyntaxTree tree);
 
         /// <summary>
-        /// Get diagnostic severity setting or a given diagnostic identifier in a given tree.
+        /// Get diagnostic severity setting for a given diagnostic identifier in a given tree.
         /// </summary>
         public abstract bool TryGetDiagnosticValue(SyntaxTree tree, string diagnosticId, out ReportDiagnostic severity);
+
+        /// <summary>
+        /// Get diagnostic severity set globally for a given diagnostic identifier
+        /// </summary>
+        public abstract bool TryGetGlobalDiagnosticValue(string diagnosticId, out ReportDiagnostic severity);
     }
 
     internal sealed class CompilerSyntaxTreeOptionsProvider : SyntaxTreeOptionsProvider
```

Comment:
❓ This type is a new public api. Should we rename it to better reflect that it's not really about TreeOptions?

Perhaps `DiagnosticSeverityProvider` although its based on top of analyzer options it really only returns `ReportDiagnostics`

Reply:
I'm not sure if there is time to rename APIs. Isn't the API locked with .NET 5 RC1 shipping?

---
In reply to: [452976584](https://github.com/dotnet/roslyn/pull/45871#discussion_r452976584) [](ancestors = 452976584)

Reply:
This better not be locked given we just had our review of APIs today and we have other things to change here too. :smile:

@chsienki Just to make sure we understood, this needs to be a different API than TryGetGlobalDiagnosticValue because _other_ options have to be slotted in the middle in terms of the precedence? And that one is just the compilation level diagnostic settings?

Reply:
`TryGetGlobalDiagnosticValue` has to exist as the user has to have a way of asking for diagnostics that still apply when not in the context of a syntax tree. 

My comment is that I think we should rename `CompilerSyntaxTreeOptionsProvider` as we only use it to get diagnostics. Though i'm now wondering if keeping as TreeOptions is better as as it allows us to add things onto it in the future

---
In reply to: [473402105](https://github.com/dotnet/roslyn/pull/45871#discussion_r473402105) [](ancestors = 473402105)

Reply:
These are not locked yet, we can still change them if needed.